torch/optim/adam.py

```diff
@@ -428,7 +428,9 @@ def _multi_tensor_adam(params: List[Tensor],
     if len(params) == 0:
         return
 
-    if capturable:
+    # If we are compiling, the compiler will determine if cudagraphs
```

Comment:
I appreciate the comment here--I wonder if there's a more discoverable place for it. You could write this once (in the first instance), and then name it as a NOTE: [compilers handle CUDAgraphs] and reference that everywhere.

I'd also appreciate knowing what the compiler would do + what checks we don't expect it to pass anymore.

Reply:
I just copied it since it's only used in two places and it's pretty short

Reply:
alright i would appreciate the comment more if it addressed why the check no longer works. my previous understanding was that it’s due to step not necessarily being on cuda when torch.compiled—is this correct?

will inductor handle moving step later? i’d rather have the comment be specific here than the vague “just let the compiler worry about it.” also, i had previously thought of the torch compiled version as a capturable implementation which is why the conflation is okay. in other words, torch compiling an optim will give u the best capturable result. will we run into issues trying to differentiate capturable when pt2 vs capturable in eager?

Reply:
Hmm, no, the check is fine, this is a performance thing again (the check is slow). The step is on cuda if we're compiling. What I mean by the check doesn't matter anymore is that if say, the step was on cuda but some param is on CPU, the optimizer will still run and be fine. Inductor will decide whether cudagraphs should be run, if there are cpu tensors it just won't run cudagraphs.